### PR TITLE
Add configurable revisions for git repositories

### PR DESCRIPTION
**Pull request by corvus-ch**
https://github.com/projectsyn/commodore/issues/198 and https://github.com/projectsyn/commodore/issues/171 asks for a way to store git revisions. This builds on top of https://github.com/projectsyn/lieutenant-operator/pull/116 to expose the new fields through the API.

## Checklist

- [x] Keep pull requests small so they can be easily reviewed.
- [x] Update the documentation.
- [x] Update the ./CHANGELOG.md.
- [x] Link this PR to related issues.
- [x] Update CRD dependencies

### CODE REVIEW

**Comment by corvus-ch** [comment]
@srueg Have I seen correctly: there are no test for the conversion between CRD and API DTOs?

And is it OK to reuse the GitRepo object here? It provides some fields not needed for this use case.

**Comment by srueg** [comment]
Yes right, there's no explicit tests for the conversion funcs. They're tested to some extent through the "service" tests tough, by using a fake K8s client (so the objects are being converted API->CRDs->API in the tests).

> And is it OK to reuse the GitRepo object here? It provides some fields not needed for this use case.

Hmm, I'd rather create a new type for this (or split the `GitRepo` type). The generated documentation gets quite confusing otherwise.

File: openapi.yaml [hunk 1]
```diff
@@ -164,6 +184,9 @@ components:
           example: |
             gitlab.com ssh-ed25519 AAAAC3NzaC1lZDI1NTE5AAAAIAfuCHKVTjquxvt6CM6tdG4SLp1Btn/nOeHHE5UOzRdf
             gitlab.com ecdsa-sha2-nistp256 AAAAE2VjZHNhLXNoYTItbmlzdHAyNTYAAAAIbmlzdHAyNTYAAABBBFSMqzJeV9rUzU4kWitGjeR4PWSa29SPqJ1fVkhtj3Hw9xjLVXVYrU9QlYWrOLXBpQ6KWjbjTDTdDkoohFzgbEY=
+        revision:
```
Comment (srueg) [inline openapi.yaml:187]: We should probably move this out of the general `GitRepo` type and create a new type (e.g. `TenantGitRepo`) based on `GitRepo` adding the `revision` field there?

**Review by srueg — CHANGES_REQUESTED** [review]
Please extend the tests to make sure the new fields get set correctly

**Review by srueg — APPROVED** [review]
Changes look good, please add some tests

**Review by srueg — APPROVED** [review]

**Pull request merged**